internal/controllers/clusterrequest_controller.go

```diff
@@ -1649,45 +1661,42 @@ func (r *ClusterRequestReconciler) handleFinalizer(
 
 // checkNodePoolProvisionStatus checks for the NodePool status to be in the provisioned state.
 func (t *clusterRequestReconcilerTask) checkNodePoolProvisionStatus(ctx context.Context,
-	nodePool *hwv1alpha1.NodePool) bool {
+	nodePool *hwv1alpha1.NodePool) (bool, bool, error) {
 
 	// Get the generated NodePool and its status.
 	exists, err := utils.DoesK8SResourceExist(ctx, t.client, nodePool.GetName(),
 		nodePool.GetNamespace(), nodePool)
 
-	if err != nil || !exists {
+	if err != nil {
```

Comment:
As err is now returned, I think we don't have to add error logs in each inner function if we are already logging the error at a the higher level.
For example, line1671-1676 and 1691-1696 and error log inside of updateHardwareProvisioningStatus

Reply:
I removed the extra error logs.